docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,351 @@
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+	    # the oidc url after it has been set up
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+```
+
+#### <a name='FederatedCredential'></a>Federated Credential
+
+The service account that the capz pod will use should be annotated with the client ID of the AAD application or User Assigned Identity and a fedrated identity should be created by associating the ServiceAccont as a subject.
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    azure.workload.identity/client-id: ${APPLICATION_CLIENT_ID:-$USER_ASSIGNED_IDENTITY_CLIENT_ID}
+  labels:
+    azure.workload.identity/use: "true"
+  name: ${SERVICE_ACCOUNT_NAME}
+  namespace: ${SERVICE_ACCOUNT_NAMESPACE}
+```
+
+Also, a federated identity should be created using the azure cli ( or via the azure portal).
+
+```bash
+az identity federated-credential create \
+  --name "kubernetes-federated-credential" \
+  --identity-name "${USER_ASSIGNED_IDENTITY_NAME}" \
+  --resource-group "${RESOURCE_GROUP}" \
+  --issuer "${SERVICE_ACCOUNT_ISSUER}" \
+  --subject "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
+```
+
+This means there are two approaches to associate the correct service account to the capz deployment.
+
+- Deploy the capz with the usual configuration YAML and then patch the service account with the client ID.
+
+- Or create the service account prior to capz deployment which uses the same service account.
+
+
+#### <a name='DistributeKeys'></a>Distribute Keys To Management Cluster
+
+Setting up credentials for management cluster which is created from the bootstrap cluster is key activity. This requires storing the keys on control plane node. One approach to do so can be the following:
+
+- A reular `clusterctl init` is done.
+- Once the cluster is up and running, the keys can be distributed on the control plane node via scp.
+- After distributing the key, the appropriate flags as it was set up for the bootstrap cluster can be setup.
+- The service account should be annotated with the client ID.
+- At this stage, `clusterctl move` can be executed.
+
+Above activities can be automated via clusterctl subcommands that is specific to azure infrastructre.
+
+### <a name='ProposedConfigurationChanges'></a>Proposed Deployment Configuration Changes
+
+- Azure Workload Identity mutating webhook should be deployed as part of capz deployment.
+
+- The capz deployment configuration should have a environment variable configuration that is used to enable/disable the use of AWI. By default it will be disabled.
+
+
+### <a name='ProposedControllerChanges'></a>Proposed Controller Changes
```

Comment:
I don't see a section about proposed API changes. This is important since currently identity is defined in the API as `identityRef`  and ` AzureClusterIdentity`

Reply:
Though I do not have any strong opinions on keeping the support for both simultaneously. Just was thinking about it and mentioned.